### DISCUSSION TITLE
sub dataset and interpolation

### DISCUSSION THREAD

**Discussion by galleon** [pydata/xarray#7106] [Q&A]
I am discovering the power of `xarray` but I am still unsure if I can do what I want.
From a GRIB file, I am reading wind profile for the whole globe at different isobaric & for a 24-hour period.
I have therefore the following:
 
<img width="720" alt="image" src="https://user-images.githubusercontent.com/259772/193078491-2d07bc33-af24-4fc4-8b11-0c5a7818a9a7.png">

My application being aeronautics, I want to compute u, v through interpolation at any given (t, alt, lat, lon) in a subset of the original data.

I know that I can do `uwind = ds.u.interp(time=t, isobaricInHPa=i, latitude=lat, longitude=lon)`

But I wanted to try to optimize space and time so  I decided to explore the following steps:

1. Get a sub dataset with all timesteps, all altitudes but a [lat_min, lat_max], [lon_min, lon_max] window
2. Move isobaric to altitude (which is more natural for my use-case)
3. Interpolate my data 

I faced newbie's issues in all my steps :-)

1. I tried this `uwind4 = uwind.where(uwind.latitude.isin([90, 89.75])).where(uwind_.longitude.isin([359.5, 359.8]))` which turned out to give me the same dataset size full of nan while I was expecting a smaller one. So I am really wondering if that is possible.

2. I tried the following
```
altitude = aero.h_isa(uwind.isobaricInhPa.data * 100) / aero.ft
uwind2 = uwind.assign_coords(altitude=("isobaricInhPa", altitude))
```
which works well but `altitude` is not an index apparently so I cannot `sel` using it. Did I miss something there ?

3. is easy (and a bit mysterious to me so blazing fast)

Thanks for your help

**Comment by TomNicholas** [comment]
Welcome to xarray!

For 1. Did you try using `.sel(latitude=[min, max], longitude=[min, max], meathod="nearest")`?

For 2. When you do that `altitude` becomes a 1d coordinate but is not backed by an index (not bold in the repr). Selection along these will become possible very soon https://github.com/pydata/xarray/issues/2028

For 3. Is there there an issue? :)

**Reply by galleon** [reply]
Thank you !

1. No I did not, how can I have missed that one. It does not do exactly what I want since it will only pick `min` and `max` values while I want all values in the intervals. I guess I will need to create from the coordinates a list of all values that are in the interval. Since all indexes seem to have comparison operators (<, ...), it would be a nice addition to the API to enable such selection ... a bit like what `where` does but just keeping the meaningful data !

2. Cool ! In the meantime is there a way to either:
  - create a new index 
  or
  - transform an index and rename it - I dont mind if I loose `isobaric` since `altitude` is just a one to one  transformation
  
  3. No ` interp` is cristal clear :-) 
  
  And thanks for the great software.

**Reply by abkfenris** [reply]
Try `.sel(latitude=slice(min, max), longitude=slice(min, max))` to get anything in the interval.

Indexes I'm less sure about.

**Reply by benbovy** [reply]
For 2, with the last Xarray release v2022.09.0 you can now do this:

```python
uwind2 = uwind.assign_coords(altitude=("isobaricInhPa", altitude)).set_xindex("altitude")

uwind2.sel(altitude=...)
```
